conda_build/index.py

```diff
@@ -56,9 +53,10 @@ def get_build_index(
     global local_output_folder
     global cached_index
     global cached_channels
-    global _channel_data
     mtime = 0
 
+    _channel_data = {}
```

Comment:
Hi @mbargull ! Wanted to ping you directly per your work in https://github.com/conda/conda-build/pull/5152 marking the `channel_data` constant for deprecation; does this update to `_channel_data` (removing the global variable and making this a local variable instead) look OK to you?

Reply:
No quite -- meaning, making it local doesn't take it far enough ;).
Sorry, I should've added a code comment there:
The idea was to remove it altogether since we populate it, but don't actually use it as of now.
Meaning, we should
1. remove everything from `expanded_channels = ...` (currently line 132) until `_channel_data["defaults"] = ...` (currently line 175), and
2. either omit the third return value or just return `None` instead.

As for point 2: I did a GitHub code search for orgs `conda`, `conda-forge`, `conda-incubator`, `bioconda`, `mamba-org` and could not find any usage of `get_build_index` outside of `conda-build` itself, so removing the return value is probably fine (returning `None` instead is a more defensive variant which wouldn't break uses of `index, ts, _ = get_build_index(...)`.

Reply:
oooh! that makes much more sense!

yeah I'd just return an empty dict then so we don't modify the return structure/type

Reply:
This would, for one, of course remove unused code, but it would also avoid iterating over the index in `expanded_channels = {rec.channel for rec in cached_index}`.
On its own this doesn't yet change much really, since the index returned by `conda.core.index.get_index`/`conda.core.index.fetch_index` already has been iterated over (`fetch_index` does `index.update((rec, rec) for rec in f.iter_records())`).
But if/when we fix that initial iteration whilst addressing gh-5154, we can finally make use of deferred package record instantiation and shave off a good 30s and a whole lot of memory consumption.
Leaving the unused `channel_data` population in place would hinder that work, of course.

Reply:
> yeah I'd just return an empty dict then so we don't modify the return structure/type

I thought about doing this too -- but I think a `None` would be a bit better since it would at least intentionally break uses of the then invalid `channel_data` (i.e., avoid silent errors in case this is really used somewhere).
N.B.: All uses in `conda-build` itself just ignore that return value anyway.